django/db/backends/base/base.py

```diff
@@ -641,3 +648,21 @@ def copy(self, alias=None, allow_thread_sharing=None):
         if allow_thread_sharing is None:
             allow_thread_sharing = self.allow_thread_sharing
         return type(self)(settings_dict, alias, allow_thread_sharing)
+
+    def _push_execute_hook(self, hook, for_many=None):
+        self.execute_hooks.append((hook, for_many))
+
+    def _pop_execute_hook(self):
+        return self.execute_hooks.pop()
+
+    @contextmanager
+    def add_execute_hook(self, hook, for_many=None):
```

Comment:
This name is a bit weird for the context manager, because it doesn't just add, but it adds and then deletes. The name `execute_hook` seems better to me:

```
with connection.execute_hook(something):
```

Reply:
Accepted